src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts

```diff
@@ -631,8 +659,15 @@ export class ReviewZoneWidget extends ZoneWidget implements ICommentThreadWidget
 			if (this._commentThread.comments!.length === 0) {
 				this.dispose();
 			}
+
+			if (deletedElementIndex === this._focusedComment) { this.setFocusedComment(undefined); }
```

Comment:
It looks like `onDidDelete` is actually never trigged from the comment node :O this is left over from a previous version of the comment API where edit and delete were explicitly implemented by the extension, now the extension just provides a list of commands. Missed removing this, sorry!

The `update` method below should get called when comments are removed

Reply:
I removed the `onDidDelete` stuff, and added more bounds checking to the setter.